action.yml

```diff
@@ -101,6 +107,22 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: "Flake.lock health check"
+      if: ${{ inputs.check-flake == 'true' }}
+      shell: bash
+      run: |
+        (
+          set -eu
+          curl --max-time 2 --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/flake-checker/stable/$RUNNER_ARCH-$RUNNER_OS > "$RUNNER_TEMP/flake-checker"
+          chmod +x "$RUNNER_TEMP/flake-checker"
+
+          if [ "${{ inputs.flake-checker-send-telemetry }}" == "false" ] || [ "${{ inputs.diagnostic-endpoint }}" != "https://install.determinate.systems/nix/diagnostic" ]; then
```

Comment:
What if users want to report their flake checker outputs to somewhere else? This seems to block them.

Reply:
We could extend the checker if we get feedback about that!

Reply:
The installer has a `diagnostic-endpoint` argument, why can't the flake checker also use that?

Reply:
No, the diagnostic endpoint is very specific to the Nix Installer's diagnostic reporting and is focused on making sure the Nix Installer works reliably and to help us fix problems.

I don't want to mix its use, because the statistics and telemetry gathered by the flake checker is more about gathering data about the world and how many potentially problematic flake.lock inputs we're identifying. They're very different motivations, and I want to keep that clearly differentiated.